doc/rdoc/markup_reference.rb

```diff
@@ -29,16 +29,31 @@
 #   see <tt>:nodoc:</tt>, <tt>:doc:</tt>, and <tt>:notnew</tt>.
 # - \RDoc directives in single-line comments;
 #   see other {Directives}[rdoc-ref:RDoc::MarkupReference@Directives].
-# - The Ruby code itself;
+# - The Ruby code itself (but not from C code);
 #   see {Documentation Derived from Ruby Code}[rdoc-ref:RDoc::MarkupReference@Documentation+Derived+from+Ruby+Code]
 #
 # == Markup in Comments
 #
-# A single-line or multi-line comment that immediately precedes
-# the definition of a class, module, method, alias, constant, or attribute
+# The treatment of markup in comments varies according to the type of file:
+#
+# - <tt>.rb</tt> (Ruby code file): markup is parsed from Ruby comments;
+#   see {Code Comments}[rdoc-ref:doc/syntax/comments.rdoc].
+# - <tt>.c</tt> (C code file): markup is parsed from C comments;
+#   see {C Comments}[https://www.w3schools.com/c/c_comments.php].
```

Comment:
I don't think it's very helpful to link to how C comments work. It would probably be more helpful to link to sections that describe how to add documentation to C files.

Reply:
Not sure what you mean.  Do you have link targets in mind?

Reply:
I couldn't find any document explaining how RDoc works for C files, but if there is one I would link to that document from here.

Reply:
Isn't _this_ the document explaining how RDoc works for C files?

Reply:
I guess what I meant is that we should link to how to use RDoc in C files. It might not be very clear to the reader how to exactly how to add documentation to methods, classes, etc. in C files. But I guess that's out of the scope of this PR. However, I still don't think it's very helpful to link how to write C or Ruby comments, the reader should at least know that much.

Reply:
Agreed.